src/Surfnet/ServiceProviderDashboard/Application/Metadata/JsonGenerator/ArpGenerator.php

```diff
@@ -64,6 +64,14 @@ public function build(MetadataConversionDto $entity)
                 if ($attr->hasMotivation()) {
                     $attributes[$urn][0]['motivation'] = $attr->getMotivation();
                 }
+
+                if ($entity->isManageEntity() && $entity->getArpAttributes()->findByUrn($urn)) {
```

Comment:
Consider using a separate method to keep the code better readable. I think it's hard to understand what is actually happening in the ArpGenerator. You could for example use `mergeWithManageAttributes` and `addManageOnlyAttributes` (for the addExtraFields). It's a suggestion but I think now is the time to refactor this to keep the code readable in the future.

Reply:
Good sugestion, will integrate them!